tools/proofers/codepoint_validator.js

```diff
@@ -58,9 +61,96 @@ $(function () {
     });
 
     $("#cc-remove").click(function () {
-        let textarea = $("#text_data");
-        textarea.val(textarea.val().replace(pattern , ""));
+        textArea.value = textArea.value.replace(badPattern , "");
         $("#checker").hide();
         $("#proofdiv").show();
     });
+
+    var above = {
+        "=": "\u0304", // macron
+        ":": "\u0308", // diaeresis
+        ".": "\u0307", // dot
+        "`": "\u0300", // grave
+        "'": "\u0301", // acute
+        "^": "\u0302", // circumflex
+        ")": "\u0306", // breve
+        "~": "\u0303", // tilde
+        ",": "\u0327", // cedilla
+        "v": "\u030C", // caron
+        "*": "\u030A", // ring
+        "(": "\u0311", // inverted breve
+    };
+
+    var below = {
+        "=": "\u0331", // macron
+        ":": "\u0324", // diaeresis
+        ".": "\u0323", // dot
+        "`": "\u0316", // grave
+        "'": "\u0317", // acute
+        "^": "\u032D", // circumflex
+        ")": "\u032E", // breve
+        "~": "\u0330", // tilde
+        ",": "\u0327", // cedilla
+        "v": "\u032C", // caron
+        "*": "\u0325", // ring
+        "(": "\u032F", // inverted breve
+    };
+
+    var ligatures = {
+        "ae": "\u00e6",
+        "AE": "\u00c6",
+        "oe": "\u0153",
+        "OE": "\u0152",
+    };
+
+    $(textArea).on("input", function() {
+        // if text changed and character before caret is ]
+        // (i.e. just inserted or pasted ] or deleted following character)
+        // check for markup like [..] for ligature or diacritical
+        // i0, i1, i3 point to these characters
+        var text = textArea.value;
+        var startPos = textArea.selectionStart;
+        var i0, i1, i3, uchar;
+
+        function substitute() {
+            // replace markup with character and mve caret back 3 places
+            textArea.value = text.slice(0, i0) + uchar + text.slice(startPos);
+            textArea.setSelectionRange(i1, i1);
+        }
+
+        i3 = startPos - 1;
+        if(text[i3] != "]") {
+            // if out of range would get ""
+            return;
+        }
+        i0 = startPos - 4;
+        if(text[i0] === "[") {
+            i1 = startPos - 3;
+            uchar = ligatures[text.slice(i1, i3)];
```

Comment:
Ligatures aren't guaranteed to be valid for the project's codepoints, we should validate those too. You could just move the validation into the substitute function.

Reply:
Good point.